core/domain/entities/custom_post_types/CustomPostTypeDefinitions.php

```diff
@@ -229,16 +229,13 @@ public function getCustomPostTypeModelNames($post_type_slug = '')
         $cpts = $this->getDefinitions();
         //first if slug passed in...
         if (! empty($post_type_slug)) {
-            //match?
+            // check that slug and cpt match
+            if (! isset($cpts[ $post_type_slug ])) {
+                return array();
+            }
             if (
-                ! isset($cpts[ $post_type_slug ])
-                || (
-                    isset($cpts[ $post_type_slug ])
-                    && (
-                        empty($cpts[ $post_type_slug ]['class_name'])
-                        && empty($cpts[ $post_type_slug ]['model_name'])
-                    )
-                )
+                empty($cpts[ $post_type_slug ]['class_name'])
```

Comment:
I'm puzzled why you split out this condition into multiple conditions?  Was it for readability? If anything, it could have been simplified to:

```php
if (
    ! isset($cpts[ $post_type_slug ])
    || (empty($cpts[ $post_type_slug ]['class_name'])
         && empty($cpts[ $post_type_slug ]['model_name'])
     )
)
```

Reply:
ya that would work too.

I split it up because it is more readable imo, and also because it means that sometimes only the first conditional will get tested, whereas using an OR means that both parts of that conditional will always get tested. So i prefer it as is now.

Reply:
I'm okay if you want to leave it as is, but just want to address this:

> whereas using an OR means that both parts of that conditional will always get tested.

That's not really accurate.   If the first part of the conditional is a match, other conditional parts will not be read, that's one of the reasons why in statements like this we always put the least expensive conditional checks first.